File: v1/examples/python/02_channels.py
"""
02_channels.py - TS-API v1 Channel

Endpoints:
  GET /api/v1/channel              - List all channels
  GET /api/v1/channel?staticSrc    - Include static stream source URLs
  GET /api/v1/channel?caps         - Include channel capabilities
  GET /api/v1/channel/status?recordingStatus - Recording status per channel
  GET /api/v1/channel/{chid}/info?caps       - Single channel capabilities

Channel fields: chid, title, displayName (NOT 'name')
"""

import os
import sys

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
from config import NVR_API_KEY
from http_client import NvrClient

if not NVR_API_KEY:
    print('NVR_API_KEY environment variable is required')
    sys.exit(1)


def main():
    client = NvrClient()
    client.set_api_key(NVR_API_KEY)

    # --- List all channels ---
    r = client.get('/api/v1/channel')
    channels = r.json()
    print(f"=== Channels ({len(channels)}) ===")
    for ch in channels:
        # Key fields: chid, title, displayName
        print(f"  chid={ch['chid']}  title={ch.get('title', '')}  "
              f"displayName={ch.get('displayName', '')}")

    # --- Static source URLs (RTMP/FLV addresses for each channel) ---
    r = client.get('/api/v1/channel?staticSrc')
    print(f"\n=== Channels with staticSrc ===")
    for ch in r.json():
        print(f"  chid={ch['chid']}  staticSrc={ch.get('staticSrc', {})}")

    # --- Channel capabilities ---
    r = client.get('/api/v1/channel?caps')
    print(f"\n=== Channels with caps ===")
    for ch in r.json():
        print(f"  chid={ch['chid']}  caps={ch.get('caps', {})}")

    # --- Recording status ---
    r = client.get('/api/v1/channel/status?recordingStatus')
    print(f"\n=== Recording Status ===")
    statuses = r.json()
    for s in statuses:
        print(f"  chid={s.get('chid')}  recording={s.get('recordingStatus')}")

    # --- Single channel capabilities (channel 1) ---
    if channels:
        chid = channels[0]['chid']
        r = client.get(f'/api/v1/channel/{chid}/info?caps')
        print(f"\n=== Channel {chid} Capabilities ===")
        print(f"  {r.json()}")


if __name__ == '__main__':
    main()

File: v1/examples/csharp/02_Channels.cs
// 02_Channels.cs - v1 API Channel List & Status Example
//
// Endpoints:
//   GET /api/v1/channel                          - Channel list
//   GET /api/v1/channel/status?recordingStatus    - Channel recording status

using System;
using System.Text.Json;
using System.Threading.Tasks;

namespace TsApiExamples.V1
{
    class Channels
    {
        static async Task Main(string[] args)
        {
            using var client = new NvrClient();
            if (string.IsNullOrEmpty(client.ApiKey))
            {
                Console.WriteLine("NVR_API_KEY environment variable is required");
                return;
            }
            client.SetApiKey(client.ApiKey);

            // ── Get channel list ──
            Console.WriteLine("=== Channel List ===");
            var r = await client.GetAsync("/api/v1/channel");
            r.EnsureSuccessStatusCode();

            using var doc = await NvrClient.ReadJsonAsync(r);
            foreach (var ch in doc.RootElement.EnumerateArray())
            {
                var chid = ch.GetProperty("chid").GetInt32();
                var title = ch.GetProperty("title").GetString();
                var displayName = ch.TryGetProperty("displayName", out var dn)
                    ? dn.GetString() : title;
                Console.WriteLine($"  CH{chid}: {displayName} (title: {title})");
            }

            // ── Get channel recording status ──
            Console.WriteLine("\n=== Channel Recording Status ===");
            var sr = await client.GetAsync("/api/v1/channel/status?recordingStatus");
            sr.EnsureSuccessStatusCode();

            using var sdoc = await NvrClient.ReadJsonAsync(sr);
            var options = new JsonSerializerOptions { WriteIndented = true };
            var pretty = JsonSerializer.Serialize(sdoc.RootElement, options);
            Console.WriteLine(pretty);
        }
    }
}
